PCAxis.Serializers/JsonStat2Serializer.cs

```diff
@@ -48,269 +33,366 @@ public class JsonStat2Serializer : IPXModelStreamSerializer
-            jsonStat.Role = new Dictionary<string, string[]>();
+            if (string.IsNullOrEmpty(variable.EliminationValue.Code)) return;
+            dimensionValue.Extension.EliminationValueCode = variable.EliminationValue.Code;
+        }
 
-			//Extension, PX 
-			if (model.Meta.InfoFile != null || model.Meta.TableID != null || model.Meta.Decimals != -1)
-			{
-				jsonStat.Extension = new Dictionary<string, object>();
-				var px = new JsonStat2.Model.Px();
+        private void AddShow(DatasetDimensionValue dimensionValue, Variable variable)
+        {
+            if (Enum.TryParse(variable.PresentationText.ToString(), out PresentationFormType presentationForm))
+            {
+                dimensionValue.Extension.Show = presentationForm.ToString().ToLower();
+            }
+        }
 
-				px.infofile = model.Meta.InfoFile;
-				px.tableid = model.Meta.TableID;
-				//If not Showdecimal has value use Decimal
-				var decimals = model.Meta.ShowDecimals < 0 ? model.Meta.Decimals : model.Meta.ShowDecimals;
-				px.decimals = decimals;
+        private void AddValueNotes(Value variableValue, Dataset dataset, DatasetDimensionValue dimensionValue)
+        {
+            if (variableValue.Notes == null) return;
 
-				jsonStat.Extension.Add(PX, px);
-			}
+            var index = 0;
+            foreach (var note in variableValue.Notes)
+            {
+                //dataset.AddValueNoteToDimension(dimensionValue, variableValue.Code, note.Mandantory, note.Text);
+                dataset.AddValueNoteToCategory(dimensionValue, variableValue.Code, note.Text);
 
+                if (note.Mandantory)
+                    dataset.AddIsMandatoryForCategoryNote(dimensionValue, variableValue.Code, index.ToString());
+                
+                index++;
+            }
+        }
 
-			if (roleTimeList.Count > 0) { jsonStat.Role.Add(TIME, roleTimeList.ToArray()); }
-            if (roleMetricList.Count > 0) { jsonStat.Role.Add(METRIC, roleMetricList.ToArray()); }
-            if (roleGeoList.Count > 0) { jsonStat.Role.Add(GEO, roleGeoList.ToArray()); }
+        private void AddVariableNotes(Variable variable, Dataset dataset, DatasetDimensionValue dimensionValue)
+        {
+            if (variable.Notes == null) return;
 
+            var noteIndex = 0;
+            foreach (var note in variable.Notes)
+            {
+                dataset.AddNoteToDimension(dimensionValue, note.Text);
 
+                if (note.Mandantory)
+                    dataset.AddIsMandatoryForDimensionNote(dimensionValue, noteIndex.ToString());
 
-            // override converter to stop adding ".0" after interger values.
-            string result = JsonConvert.SerializeObject(jsonStat,new DecimalJsonConverter());
- 
-            return result;
+                noteIndex++;
+            }
         }
-        
 
-        public void Serialize(PXModel model, Stream stream)
+        private void AddContacts(Dataset dataset, string contactString)
```

Comment:
We do a slightly diffrent mapping of the contact information in the custom json format https://github.com/statisticssweden/PxWeb/blob/860c2e35124d684e6ec833d8778a0ee83403325e/PxWeb/Mappers/TableMetadataResponseMapper.cs#L254 where we utilyze the a phared contect object if it exists. (Only exists when CNMM and PX files that that have the the same format as the SQL-Parser.)

Reply:
Changed how contact is set. It now uses Paxiom.Contact if it exist. 
Multiple contacts defined in ContInfo.Contact will be splitted into seperate contacts in root extenion/contact.